### PR TITLE
chore: Detect client disconnect in pg proxy

### PR DESCRIPTION
**Pull request by scsmithr**
- Lowers log level for unexpected eof errors (client disconnects).
- Raises log level for all other errors

Closes https://github.com/GlareDB/glaredb/issues/2255

### CODE REVIEW

**Review by greyscaled — APPROVED** [review]

**Comment by greyscaled** [comment]
can we mark this as closing the issue #2255 ?

**Comment by scsmithr** [comment]
> can we mark this as closing the issue #2255 ?

done

File: crates/glaredb/src/pg_proxy.rs [hunk 1]
```diff
@@ -85,7 +87,9 @@ impl PgProxy {
                         debug!("client connected (proxy)");
                         match handler.proxy_connection(inbound).await {
                             Ok(_) => debug!("client disconnected"),
-                            Err(e) => debug!(%e, "client disconnected with error."),
+                            Err(PgSrvError::Io(e)) if e.kind() == io::ErrorKind::UnexpectedEof =>
+                                trace!("client disconnected"),
```
Comment (tychoish) [inline crates/glaredb/src/pg_proxy.rs:91]: should the `Ok()` case also be a trace then? should we distinguish between the two?

Comment (scsmithr) [reply crates/glaredb/src/pg_proxy.rs:91]: will add

Comment (scsmithr) [reply crates/glaredb/src/pg_proxy.rs]: Downgraded debugs to trace, and used different messages in both cases (in case we need to distinguish).

**Pull request merged**